third-party/realdds/src/topics/image-msg.cpp

```diff
@@ -34,6 +79,38 @@ image::create_topic( std::shared_ptr< dds_participant > const & participant, cha
 }
 
 
+/*static*/ bool
+image::take_next( dds_topic_reader & reader, image * output, eprosima::fastdds::dds::SampleInfo * info )
+{
+    raw::device::image raw_data;
+    eprosima::fastdds::dds::SampleInfo info_;
+    if ( !info )
+        info = &info_;  // use the local copy if the user hasn't provided their own
+    auto status = reader->take_next_sample( &raw_data, info );
+    if ( status == ReturnCode_t::RETCODE_OK )
+    {
+        // We have data
+        if ( output )
+        {
+            // Only samples for which valid_data is true should be accessed
+            // valid_data indicates that the instance is still ALIVE and the `take` return an
+            // updated sample
+            if ( !info->valid_data )
+                output->invalidate();
+            else
+                *output = raw_data; //TODO - optimize copy, use dds loans
```

Comment:
I think we have copies all over and it's hard to track.
You can please spot every copy in the image flow and mark it with a similar comment?
This way we can later list all copies at a single search and optimize where we can.
Thanks :)

Reply:
We have 2 copies. one here (that can be avoided later by loaning the sample) and a second in context.cpp when copying to LibRS frame type.

Reply:
`take_next` is another copy no?

Reply:
This should be `*output = std::move( raw_data )`.

Reply:
Fixed